### PR TITLE
Feature/textbox context menu

### PR DESCRIPTION
**Pull request by danwalmsley**
## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

File: src/Avalonia.Themes.Fluent/TextBox.xaml [hunk 1]
```diff
@@ -28,6 +33,7 @@
     <Setter Property="MinWidth" Value="{DynamicResource TextControlThemeMinWidth}" />
     <Setter Property="Padding" Value="{DynamicResource TextControlThemePadding}" />
     <Setter Property="FocusAdorner" Value="{x:Null}" />
+    <Setter Property="ContextMenu" Value="{DynamicResource DefaultTextBoxContextMenu}" />
```
Comment (maxkatz6) [inline src/Avalonia.Themes.Fluent/TextBox.xaml:36]: Why we can't use StaticResource here? I don't think "DefaultTextBoxContextMenu" resources is supposed to be changed in runtime.

**Comment by maxkatz6** [comment]
Putting here UWP TextBox API for cut/copy/paste operations.
Note, there is no any CanCut or CanCopy, but only CanPaste. Probably API assumes developer to check for null or empty manually.

```csharp
public void PasteFromClipboard();
public void CopySelectionToClipboard();
public void CutSelectionToClipboard();
public bool CanPasteClipboardContent { get; }

public event TextControlPasteEventHandler Paste;
public event TypedEventHandler<TextBox, TextControlCopyingToClipboardEventArgs> CopyingToClipboard;
public event TypedEventHandler<TextBox, TextControlCuttingToClipboardEventArgs> CuttingToClipboard
```

**Comment by maxkatz6** [comment]
Not sure about methods naming, but events could be useful to have.
Just checked, all of them (Paste, CopyingToClipboard and CuttingToClipboard) has EventArgs with Handled property, that allows to cancel operations from code behind.

File: src/Avalonia.Themes.Fluent/TextBox.xaml [hunk 2]
```diff
@@ -14,6 +14,11 @@
...
+    <ContextMenu x:Key="DefaultTextBoxContextMenu">
+      <MenuItem Header="Cut" Command="{Binding $parent[TextBox].Cut}" IsEnabled="{Binding $parent[TextBox].CanCut}" InputGesture="{x:Static TextBox.CutGesture}" />
```
Comment (Deadpikle) [inline src/Avalonia.Themes.Fluent/TextBox.xaml:18]: Feature request: In my app, I hacked up a Cut/Copy/Paste menu for `TextBox`, but I also have an icon for this menu. Could you add a way to configure the icon (through a resource or binding or some other way) for each of these menu options? Thanks!

(P.S. This is going to be great to have built-in to Avalonia 😁)

**Review by jmacato — APPROVED** [review]
LGTM!

**Pull request merged**